src/XMakeTasks/AssemblyDependency/AssemblyFoldersFromConfig/AssemblyFoldersFromConfig.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Shared;
+using ProcessorArchitecture = System.Reflection.ProcessorArchitecture;
+
+namespace Microsoft.Build.Tasks.AssemblyFoldersFromConfig
+{
+    internal class AssemblyFoldersFromConfig : IEnumerable<AssemblyFoldersFromConfigInfo>
+    {
+        /// <summary>
+        /// The list of directory names found from the config file.
+        /// </summary>
+        private readonly List<AssemblyFoldersFromConfigInfo> _directoryNames = new List<AssemblyFoldersFromConfigInfo>();
+
+        /// <summary>
+        /// Construct.
+        /// </summary>
+        /// <param name="targetRuntimeVersion">The runtime version property from the project file.</param>
+        internal AssemblyFoldersFromConfig(string configFile, string targetRuntimeVersion, ProcessorArchitecture targetArchitecture)
+        {
+            if (!FileUtilities.FileExistsNoThrow(configFile))
+            {
+                return;
+            }
+
+            var collection = AssemblyFolderCollection.Load(configFile);
+            var assemblyTargets = GatherVersionStrings(targetRuntimeVersion, collection);
+
+            bool is64BitOs = Environment.Is64BitOperatingSystem;
+            bool targeting64Bit = targetArchitecture == ProcessorArchitecture.Amd64 ||
+                                  targetArchitecture == ProcessorArchitecture.IA64;
```

Comment:
Ah, this is pretty confusing, but this shouldn't be needed here. It's in `AssemblyFoldersEx` because you have to look in the WOW registry hive explicitly depending on your own bitness, but that won't be true of the config file.